interop-tests/src/arch.rs

```diff
@@ -47,87 +36,131 @@ pub(crate) mod native {
+    ) -> Result<(Swarm<B>, String)> {
+        let (swarm, addr) = match (
+            transport,
+            from_env::<SecProtocol>("security"),
+            from_env::<Muxer>("muxer"),
+        ) {
+            (Transport::QuicV1, _, _) => {
+                let swarm = libp2p::SwarmBuilder::with_new_identity()
+                    .with_tokio()
+                    .with_quic()
+                    .with_behaviour(behaviour_constructor)?
+                    .build();
+                (swarm, format!("/ip4/{ip}/udp/0/quic-v1"))
+            }
+            (Transport::Tcp, Ok(SecProtocol::Tls), Ok(Muxer::Yamux)) => {
+                let swarm = libp2p::SwarmBuilder::with_new_identity()
+                    .with_tokio()
+                    .with_tcp(
+                        tcp::Config::default(),
+                        tls::Config::new,
+                        yamux::Config::default,
+                    )?
+                    .with_behaviour(behaviour_constructor)?
+                    .build();
+                (swarm, format!("/ip4/{ip}/tcp/0"))
+            }
+            (Transport::Tcp, Ok(SecProtocol::Tls), Ok(Muxer::Mplex)) => {
+                let swarm = libp2p::SwarmBuilder::with_new_identity()
+                    .with_tokio()
+                    .with_tcp(
+                        tcp::Config::default(),
+                        tls::Config::new,
+                        mplex::MplexConfig::default,
+                    )?
+                    .with_behaviour(behaviour_constructor)?
+                    .build();
+                (swarm, format!("/ip4/{ip}/tcp/0"))
+            }
+            (Transport::Tcp, Ok(SecProtocol::Noise), Ok(Muxer::Yamux)) => {
+                let swarm = libp2p::SwarmBuilder::with_new_identity()
+                    .with_tokio()
+                    .with_tcp(
+                        tcp::Config::default(),
+                        noise::Config::new,
+                        yamux::Config::default,
+                    )?
+                    .with_behaviour(behaviour_constructor)?
+                    .build();
+                (swarm, format!("/ip4/{ip}/tcp/0"))
+            }
+            (Transport::Tcp, Ok(SecProtocol::Noise), Ok(Muxer::Mplex)) => {
+                let swarm = libp2p::SwarmBuilder::with_new_identity()
+                    .with_tokio()
+                    .with_tcp(
+                        tcp::Config::default(),
+                        noise::Config::new,
+                        mplex::MplexConfig::default,
+                    )?
+                    .with_behaviour(behaviour_constructor)?
+                    .build();
+                (swarm, format!("/ip4/{ip}/tcp/0"))
+            }
+            (Transport::Ws, Ok(SecProtocol::Tls), Ok(Muxer::Yamux)) => {
+                let swarm = libp2p::SwarmBuilder::with_new_identity()
+                    .with_tokio()
+                    .with_websocket(libp2p_tls::Config::new, yamux::Config::default)
```

Comment:
Why can't this just be called via the module that is imported?

```suggestion
                    .with_websocket(tls::Config::new, yamux::Config::default)
```

Reply:
:+1: 5a8c85393